tests/_test_process_executor.py

```diff
@@ -704,6 +705,12 @@ def test_no_stale_references(self):
 
         collected = False
         for _ in range(5):
+            if bool(sysconfig.get_config_var("Py_GIL_DISABLED")):
+                # XXX: full GC collect is needed on free-threading CPython to
```

Comment:
I seem to vaguely remember some differences in the GC under free-threading, there maybe more relevant details [there](https://github.com/python/cpython/blob/main/InternalDocs/garbage_collector.md#gc-for-the-free-threaded-build)

Reply:
By the way, just out of curiosity, what was the issue fixed by this change?

Not sure about the behaviour here, but you may want to consider using `not sys._is_gil_enabled()` to check sure the GIL is not enabled at run time? `sysconfig.get_config_var("Py_GIL_DISABLED")` is build-time information (i.e. this is a free-threaded build but you can still use `PYTHON_GIL=1` with a free-threaded build to enable the GIL).

Reply:
Indeed, the runtime condition sounds more accurate.